jetty-maven/jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/AbstractWebAppMojo.java

```diff
@@ -0,0 +1,463 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.maven.plugin;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.InvalidPluginDescriptorException;
+import org.apache.maven.plugin.MavenPluginManager;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.PluginDescriptorParsingException;
+import org.apache.maven.plugin.PluginManagerException;
+import org.apache.maven.plugin.PluginResolutionException;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.classworlds.realm.ClassRealm;
+import org.codehaus.plexus.configuration.PlexusConfiguration;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.jetty.maven.MavenServerConnector;
+import org.eclipse.jetty.maven.ScanTargetPattern;
+import org.eclipse.jetty.maven.utils.MavenProjectHelper;
+import org.eclipse.jetty.server.RequestLog;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.resource.Resource;
+import org.twdata.maven.mojoexecutor.MojoExecutor;
+
+/**
+ * AbstractWebAppMojo
+ *
+ * Base class for common behaviour of jetty mojos.
+ */
+public abstract class AbstractWebAppMojo extends AbstractMojo
+{
+
+    @Inject
+    protected BuildPluginManager buildPluginManager;
+
```

Comment:
All of the parameters that are declared in this class are duplicated in each of the `ee8/9/10` classes also called `AbstractWebAppMojo` - that seems undesirable because: a) duplication is bad b) you can't add any new parameter to the `ee` specific versions of this class because it's unconfigurable. 

I still think there's something wrong with the inheritance hierarchy because ISTM more natural that the `ee` specific classes would inherit off these common base classes rather than have nothing to do with them.

Reply:
> All of the parameters that are declared in this class are duplicated in each of the `ee8/9/10` classes also called `AbstractWebAppMojo` - that seems undesirable because: a) duplication is bad b) you can't add any new parameter to the `ee` specific versions of this class because it's unconfigurable.
> 

I explained in this comment: https://github.com/eclipse/jetty.project/pull/9216#discussion_r1091387036

> I still think there's something wrong with the inheritance hierarchy because ISTM more natural that the `ee` specific classes would inherit off these common base classes rather than have nothing to do with them.

what could be so specific for a user? 
We are talking here about a development maven plugin users just neeed to run a webapp module in development not something used in production. 
In case a user really need something specific, he will write a some `jetty-env.xml` or `jetty-webapp.xml` because he will need it anyway for deploying in a jetty distribution.
I would prefer we provide a minimum of usable options for users to configure in the pom (we can't even configure the port in the pom and we are talking about specific eeX configuration maybe used by few users)